### DISCUSSION TITLE
ClearScript for game engine?

### DISCUSSION THREAD

**Discussion by bangclash** [ClearFoundry/ClearScript#326] [General]
I'm building a game engine (using MonoGame) and you would to like allow for modding through scripting. ClearScript seems like a great solution. ClearScript almost seems too good to be true. Has anyone done this or does it seem like a good idea?

**Comment by ClearScriptLib** [comment]
Hi @bangclash,

Thanks for your interest in ClearScript!

As with most software, ClearScript comes with tradeoffs and caveats. Although it provides easy access to V8, a JavaScript engine with elite performance and standards support, the boundary between managed and native code is a significant bottleneck.

We therefore recommend ClearScript for applications that host _substantial_ amounts of script code, especially if some of it is performance-critical. In such applications, the host API should be designed carefully to minimize hops across that expensive boundary.

If on the other hand your application runs smaller bits of script code that depend on chatty interaction with the host, then a purely managed solution such as [Jint](https://github.com/sebastienros/jint) might be a better choice.

To be honest though, performance has never been ClearScript's primary goal. Instead, our aim is to make it as easy as possible to add standards-based scripting to _existing_ .NET applications. That's why ClearScript goes to great lengths to make as many CLR features as possible accessible to script code.

Good luck!
